### PR TITLE
[ISSUE #294] Resolve problem print error log when producer/consumer shutdown

### PR DESCRIPTION
**Pull request by wenfengwang**
fix https://github.com/apache/rocketmq-client-go/issues/294

### CODE REVIEW

**Comment by xujianhai666** [comment]
> fix #294

@wenfengwang ci failed.

File: internal/remote/tcp_conn.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+func (pool *tcpConnPool) putBack(wrapper tcpConnWrapper) {
+	pool.mutex.Lock()
+	defer pool.mutex.Unlock()
+
+}
+
+type tcpConnWrapper struct {
+	net.Conn
+	closed bool
```
Comment (xujianhai666) [inline internal/remote/tcp_conn.go]: never use bool directly, use  Bool in "go.uber.org/atomic"

File: internal/remote/tcp_conn.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	"net"
+	"sync"
+)
+
+var (
+// TODO: Adding TCP Connections Pool, https://github.com/apache/rocketmq-client-go/issues/298
+//pool = &tcpConnPool{}
+)
+
+type tcpConnPool struct {
```
Comment (xujianhai666) [inline internal/remote/tcp_conn.go]: could move this code to antoher commit?

Comment (wenfengwang) [reply internal/remote/tcp_conn.go]: done

File: internal/remote/tcp_conn.go (hunk 1 above)
Comment (wenfengwang) [reply internal/remote/tcp_conn.go]: done

**Pull request merged**